### PR TITLE
Update `PdosWorkChain` and tests

### PR DESCRIPTION
**Pull request by mbercx**
Besides some minor changes, this PR does the following:

* Add the `get_builder_from_protocol()`  method.
* Include the `pw.parameters` input for the `nscf` namespace and overwrite
  the default parameters with the provided inputs.
* Instead of making `Emin` and `Emax` required, use the minimum and maximum
  band energy from the `nscf` calculation in case they aren't provided.
* Use the `out_many` method for exposing the outputs.
* Update the tests for the `PdosWorkChain`.

TODO:

- [x] Maybe it makes more sense to put the `PdosWorkChain` in a `pdos.py` module instead of a separate directory and a `base.py` module? It's not a `BaseRestartWorkChain`, and I don't think we'll be adding more work chains that should be placed in the `pdos` directory.
- [x] Currently the inputs for the `nscf` calculation are provided by the `base` inputs and overridden by the `nscf` inputs. I was considering separating the `base` and `nscf` inputs entirely, as we do for the `final_scf` of the `PwRelaxWorkChain`. The main reason is simplicity and consistency. Since we now have the `get_builder_from_protocol` method, it's no longer such a pain to provide the same inputs twice. However, if the user wants to change anything, it's still tedious (and possibly error-prone) that they would have to change the input in two namespaces. On the other hand, in case they _only_ want to change an input for the `scf`, they now have to change this setting in the `base` input, and then set the default for the `nscf`. Not sure if there is a use case for this. A similar question also arises for the `parameters`/`dos`/`projwfc` inputs.
- [x] Currently the `nscf` inputs are set to defaults in the `run_scf` step. Maybe it would be better to set these defaults in the protocol `yaml` file?
- [ ] Think about good protocol defaults, e.g. for the `DeltaE` input, or the `kpoints_distance`.
- [ ] Add more validation? Think if there are inputs that need to be the same for the `'scf'` and `'nscf'` calculations.

### CODE REVIEW

File: aiida_quantumespresso/workflows/pdos.py [hunk 1]
```diff
@@ -155,45 +174,49 @@ def clean_workchain_calcs(workchain):
...
+            exclude=('clean_workdir', 'pw.structure', 'pw.parent_folder'),
+            # include=('kpoints', 'kpoints_distance', 'kpoints_force_parity', 'pw.metadata.options', 'pw.parameters'),
             namespace_options={
-                'help': 'Optional inputs for nscf calculation, that override those set in the `base` namespace'})
-        spec.input('nscf.pw.metadata.options', valid_type=dict, required=False)
-
-        # DOS & PDOS
+                'help': 'Optional inputs for `nscf` calculation, that override those set in the `base` namespace.',
+                'validator': validate_base_nscf
+            }  # TODO optionally allow PwCalculation output parent_folder? # pylint: disable=fixme
```
Comment (mbercx) [inline aiida_quantumespresso/workflows/pdos.py]: I guess the idea would be to allow the user to specify a parent folder for the `'nscf'` calculation, and skip the `'scf'`  one? (that's why I moved this todo here) In this case the `base_scf` input will become optional, and we can add validation at the top level to see that either the `base_scf` or `base_nscf.pw.parent_folder` inputs are provided.

File: tests/workflows/test_pdos.py [hunk 2]
```diff
@@ -105,35 +67,44 @@ def test_default(
     # TODO ensure this test fails if the output link from PwCalculation changes from `output_parameters` # pylint: disable=fixme
```
Comment (mbercx) [inline tests/workflows/test_pdos.py:67]: Why is this added test important? The output link for the `PwCalculation` is unlikely to change, no?

**Comment by mbercx** [comment]
@chrisjsewell this is ready for review! I've left some comments re TODO's that are still left, and we still need to figure out reasonable defaults for the protocol and maybe add some more validation for inputs that need to be the same for the `scf` and `nscf`.

Currently the `nscf` also has a bunch of parameters set which don't make any sense (but don't have an influence on the calculation, I think). I could `pop` them all in the `get_builder_from_protocol` method, but I'm thinking it would be better to allow us to specify that a parameter should not be set by setting it to `null` in the `.yaml` file. I'll do that in a separate PR, though.

File: tests/workflows/test_pdos.py (hunk 2 above)
Comment (chrisjsewell) [reply tests/workflows/test_pdos.py:67]: Yeh it was on the cards at some point to remove the `output_` prefix, because really its duplication having `node.outputs.output_parameters` and better to just have `node.outputs.parameters`. But obviously that hasn't changed as yet

File: aiida_quantumespresso/workflows/pdos.py (hunk 1 above)
Comment (chrisjsewell) [reply aiida_quantumespresso/workflows/pdos.py]: yep, obviously you may have already run a bunch of scf/relax calculations and don't want to have to repeat them to generate the PDOS

File: tests/workflows/test_pdos.py (hunk 2 above)
Comment (mbercx) [reply tests/workflows/test_pdos.py:67]: I've removed the TODO for now. I do agree that prepending `output` to the outputs (also for the `output_structure` for example) is a bit redundant, but perhaps it's not really critical to make this (backwards-incompatible) change. I'll open an issue for this if there isn't already one where we can discuss this.

File: aiida_quantumespresso/workflows/pdos.py (hunk 1 above)
Comment (mbercx) [reply aiida_quantumespresso/workflows/pdos.py]: I've updated the work chain to make the `base_scf` optional in dd34b5a.

**Comment by mbercx** [comment]
Thanks for the comments, @chrisjsewell! Would be great if you could give this PR a proper review soon, so we can merge it into `develop`. Will have a meeting later today with @csadorf to discuss integrating the work chain in the QE app.

**Review by chrisjsewell — APPROVED** [review]
